### PR TITLE
Fix output of cookbook list/show

### PR DESCRIPTION
**Pull request by aknarts**
## Description
This PR turns the output from the following:
```json
$ knife cookbook list -a -F json
[
  "7-zip                       1.0.2",
  "api_docs                    0.1.0",
  "apt                         1.1.1",
  "ark                         4.0.0  3.1.1  0.9.0  0.3.0",
  "audit-cis                   0.4.0",
  "authenticator               0.1.5  0.1.4",
  "aws-static-ip-ha            0.1.0",
  "aws-tools                   0.0.5"
]
```
to this
```json
$ knife cookbook list -a -F json
{
  "7-zip": [
    "1.0.2"
  ],
  "api_docs": [
    "0.1.0"
  ],
  "apt": [
    "1.1.1"
  ],
  "ark": [
    "4.0.0",
    "3.1.1",
    "0.9.0",
    "0.3.0"
  ],
  "audit-cis": [
    "0.4.0"
  ],
  "authenticator": [
    "0.1.5",
    "0.1.4"
  ],
  "aws-static-ip-ha": [
    "0.1.0"
  ],
  "aws-tools": [
    "0.0.5"
  ]
}
```

This has a potential of breaking peoples script if they adjusted to the broken format, that is why this is marked as a breaking change.

This brings the output of the following in line with other listings coming out of chef

#### knife cookbook show
summary(default)
```
$ knife cookbook show ark
ark: 4.0.0 3.1.1 0.9.0 0.3.0
```
text
```
$ knife cookbook show ark -F text
ark:
  4.0.0
  3.1.1
  0.9.0
  0.3.0
```
json
```json
$ knife cookbook show ark -F json
{
  "ark": [
    "4.0.0",
    "3.1.1",
    "0.9.0",
    "0.3.0"
  ]
}
```
yaml
```yaml
$ knife cookbook show ark -F yaml
---
ark:
- 4.0.0
- 3.1.1
- 0.9.0
- 0.3.0
```
pp
```
$ knife cookbook show ark -F pp
{"ark"=>["4.0.0", "3.1.1", "0.9.0", "0.3.0"]}
```
#### knife cookbook list
summary(default)
```
$ knife cookbook list -a
7-zip:                     1.0.2
api_docs:                  0.1.0
apt:                       1.1.1
ark:                       4.0.0 3.1.1 0.9.0 0.3.0
audit-cis:                 0.4.0
auditbeat:                 0.0.42
authenticator:             0.1.5 0.1.4
aws-static-ip-ha:          0.1.0
aws-tools:                 0.0.5
```
text
```
$ knife cookbook list -a -F text
7-zip:                     1.0.2
api_docs:                  0.1.0
apt:                       1.1.1
ark:
  4.0.0
  3.1.1
  0.9.0
  0.3.0
audit-cis:                 0.4.0
auditbeat:                 0.0.42
authenticator:
  0.1.5
  0.1.4
aws-static-ip-ha:          0.1.0
aws-tools:                 0.0.5
```
json
```json
$ knife cookbook list -a -F json
{
  "7-zip": [
    "1.0.2"
  ],
  "api_docs": [
    "0.1.0"
  ],
  "apt": [
    "1.1.1"
  ],
  "ark": [
    "4.0.0",
    "3.1.1",
    "0.9.0",
    "0.3.0"
  ],
  "audit-cis": [
    "0.4.0"
  ],
  "authenticator": [
    "0.1.5",
    "0.1.4"
  ],
  "aws-static-ip-ha": [
    "0.1.0"
  ],
  "aws-tools": [
    "0.0.5"
  ],
}
```
yaml
```yaml
$ knife cookbook list -a -F yaml
---
7-zip:
- 1.0.2
api_docs:
- 0.1.0
apt:
- 1.1.1
ark:
- 4.0.0
- 3.1.1
- 0.9.0
- 0.3.0
audit-cis:
- 0.4.0
authenticator:
- 0.1.5
- 0.1.4
aws-static-ip-ha:
- 0.1.0
aws-tools:
- 0.0.5
```
pp
```
$ knife cookbook list -a -F pp
{"7-zip"=>["1.0.2"],
 "api_docs"=>["0.1.0"],
 "apt"=>["1.1.1"],
 "ark"=>["4.0.0", "3.1.1", "0.9.0", "0.3.0"],
 "audit-cis"=>["0.4.0"],
 "authenticator"=>["0.1.5", "0.1.4"],
 "aws-static-ip-ha"=>["0.1.0"],
 "aws-tools"=>["0.0.5"],
}
```

## Related Issue
This is a PR for #13653

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by williamtheaker** [comment]
I think this closes https://github.com/chef/chef/issues/10404

**Review by jaymzh — CHANGES_REQUESTED** [review]
For the text version, lets keep the nice justification.

For the JSON, lets pretty-generate it

File: knife/lib/chef/knife/core/generic_presenter.rb [hunk 1]
```diff
@@ -206,26 +206,32 @@ def extract_nested_value(data, nested_value_spec)
...
+          else
+            case parse_format_option
+              when :summary
+                cookbooks = {}
+                versions_by_cookbook.map do |cookbook, versions|
+                  cookbooks[cookbook] = versions.join(" ")
+                end
+                cookbooks
+              else
+                versions_by_cookbook
```
[the remark this answers is not in the document]
Comment (aknarts) [reply knife/lib/chef/knife/core/generic_presenter.rb:231]: I am sorry but that makes the output way worse...
Your changes produce this:
Screenshot from 2023-04-12 11-21-23
Screenshot from 2023-04-12 11-21-10

My original changes produce this:
Screenshot from 2023-04-12 11-22-08
Screenshot from 2023-04-12 11-21-48

It is still formatted properly in the summary section and json is not polluted by things that do not belong to json

**Review by jaymzh — APPROVED** [review]
Ah, because `ui.output` which is where this output goes, does all the formatting for us! I didn't know that (sorry, I'm not as familiar with knife). This looks great.

we SHOULD note that if anyone is parsing the output of `knife cookbook list`, this will bite them, but I think that's OK, that's not an API.

**Pull request merged**